flag-gen/assets_generate.go

```diff
@@ -0,0 +1,53 @@
+// +build ignore
+
+package main
+
+import (
+	"github.com/shurcooL/httpfs/union"
+	"github.com/shurcooL/vfsgen"
+	"log"
+	"net/http"
+	"os"
+	"time"
+)
+
+// zeroModTimeFileSystem is an http.FileSystem wrapper.
+// It exposes a filesystem exactly like Source, except
+// all file modification times are changed to zero.
+// See https://github.com/shurcooL/vfsgen/pull/40#issuecomment-355416103
+type zeroModTimeFileSystem struct {
+	Source http.FileSystem
+}
+
+func (fs zeroModTimeFileSystem) Open(name string) (http.File, error) {
+	f, err := fs.Source.Open(name)
+	return file{f}, err
+}
+
+type file struct {
+	http.File
+}
+
+func (f file) Stat() (os.FileInfo, error) {
+	fi, err := f.File.Stat()
+	return fileInfo{fi}, err
+}
+
+type fileInfo struct {
+	os.FileInfo
+}
+
+func (fi fileInfo) ModTime() time.Time { return time.Time{} }
```

Comment:
I still disagree with this and I have not heard an good argument why we need to bodge normal filesystem operations.

Reply:
I hear you. Will remove this in the next commit

Reply:
Hey @AudriusButkevicius 

I found a valid argument for this point. Please check `assets_vsfdata.go` file. The issue here is that on regeneration during builds, the mod time of the file will change. This will reflect inside `assets_vsfdata.go`. In each build, we check if there is any diff in between newly generated files and previous files. This change in modtime will cause a git diff everytime a build is run causing our build to fail.

Reply:
![mtime](https://user-images.githubusercontent.com/25657923/62425261-c5e77080-b6f6-11e9-8d76-6b721354cede.jpg)

Check this out

Reply:
@AudriusButkevicius ping

Reply:
the thing is when we do a git clone, the mod time changes. this prompts the vfsgen plugin to generate new assets file as it perceives the files as changed. hence the need for the zero mod time fs work around.

Reply:
Fine